content/drone-plugins/drone-s3-cache/index.md

```diff
@@ -0,0 +1,174 @@
+---
+date: 2017-08-29T00:00:00+00:00
+title: Amazon S3 Cache
+author: drone-plugins
+tags: [ amazon, aws, s3, cache ]
+repo: drone-plugins/drone-s3-cache
+logo: amazon_s3.svg
+image: plugins/s3-cache
+---
+
+The S3 cache plugin can be used to preserve files and directories between builds. The below pipeline configuration demonstrates simple usage:
+
+```yaml
+pipeline:
+  restore-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    restore: true
+
+  build:
+    image: node
+    commands:
+      - npm install
+
+  rebuild-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    rebuild: true
+    mount:
+      - node_modules
```

Comment:
The rebuild-cache step should only happen on a `push` event.

For `pull_request` you could end up adding cache entries into the target branch. For `tag` its never going to be used again. For `deployment` same thing.

Reply:
Added a push event condition.